app/cloud.go

```diff
@@ -62,57 +121,146 @@ func (a *App) NotifySystemAdminsToUpgrade(c *request.Context, currentUserTeamID
 		return appErr
 	}
 
+	now := model.GetMillis()
+
+	data, err := a.Srv().Store.NotifyAdmin().Get(trial)
+	if err != nil {
+		return model.NewAppError("SendNotifyAdminPosts", "app.notify_admin.send_notification_post.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	data = filterNotificationData(data, func(nad *model.NotifyAdminData) bool { return string(nad.RequiredPlan) != currentSKU })
+
+	if len(data) == 0 {
+		mlog.Warn("No notification data available")
+		return nil
+	}
+
+	userBasedData := a.groupNotifyAdminByUser(data)
+	featureBasedData := a.groupNotifyAdminByFeature(data)
+	props := make(model.StringInterface)
+
 	for _, admin := range sysadmins {
 		T := i18n.GetUserTranslations(admin.Locale)
+		message := T("app.cloud.upgrade_plan_bot_message", map[string]interface{}{"UsersNum": len(userBasedData), "WorkspaceName": workspaceName})
```

Comment:
Left a comment in Figma to see if we can drop the workspace name so we don't have to be cloud specific